classes/class-swsales-reports.php

```diff
@@ -42,141 +73,263 @@ public static function show_reports_page() { ?>
-					<p>
-						<?php
-							printf(
-								wp_kses_post( '<a href="%s" title="%s">Landing</a> Page Visits', 'sitewide-sales' ),
-								get_permalink( $sitewide_sale->get_landing_page_post_id() ),
-								get_the_title( $sitewide_sale->get_landing_page_post_id() )
-							);
-						?>
-					</p>
-				</div>
-				<div id="swsales_reports-data-section_sales" class="swsales_reports-data-section">
-					<h1><?php echo esc_attr( $sitewide_sale->get_checkout_conversions() ); ?></h1>
-					<p>
-						<?php
-							printf(
-								wp_kses_post( apply_filters( 'swsales_checkout_conversions_title', __( 'Checkout Conversions', 'sitewide-sales' ), $sitewide_sale ) )
-							);
-						?>
-					</p>
-				</div>
-				<div class="swsales_reports-data-section">
-					<h1><?php echo esc_attr( $sitewide_sale->get_revenue() ); ?></h1>
-					<p><?php esc_html_e( 'Sale Revenue', 'sitewide-sales' ); ?></p>
-				</div>
-			</div>
-			<?php
-			// Daily Revenue Chart.
-			// Build an array with each day of sale as a key to store revenue data in.
-			$date_array_all = array();
-			$period = new \DatePeriod(
-				new \DateTime( $sitewide_sale->get_start_date( 'Y-m-d' ) ),
-				new \DateInterval('P1D'),
-				new \DateTime( $sitewide_sale->get_end_date( 'Y-m-d' ) . ' + 1 day' )
-			);
-			foreach ($period as $key => $value) {
-				$date_array_all[ $value->format('Y-m-d') ] = 0.0;
+
+		// Bail if given elements aren't SWSales_Sitewide_Sale objects.
+		foreach ( $sitewide_sales as $sitewide_sale ) {
+			if ( ! is_a( $sitewide_sale, 'Sitewide_Sales\classes\SWSales_Sitewide_Sale' ) ) {
+				return;
 			}
+		}
+		?>
+		<div id="swsales_overall_sale_performance" class="swsales_reports-box">
+			<h2 class="swsales_reports-box-title"><?php esc_html_e( 'Overall Sale Performance', 'sitewide-sales' ); ?></h2>
+			<table>
+				<thead>
+					<tr>
+						<th width="25%"></th>
+						<th><?php esc_html_e( 'Banner Reach', 'sitewide-sales' ); ?></th>
+						<th><?php esc_html_e( 'Landing Page Visits', 'sitewide-sales' ); ?></th>
+						<th><?php esc_html_e( 'Conversions', 'sitewide-sales' ); ?></th>
+						<th><?php esc_html_e( 'Sale Revenue', 'sitewide-sales' ); ?></th>
+					</tr>
+				</thead>
+				<tbody>
+				<?php
+					$daily_chart_data = [];
+					$max_sale_days = 0;
+					foreach ( $sitewide_sales as $key => $sitewide_sale ) {
+						$diff_rate = null;
+						if ( count( $sitewide_sales ) > 1 && $key === 0 ) {
+							$diff_rate = self::build_diff_rate_array( $sitewide_sale, $sitewide_sales[1]  );
+						}
+					?>
+					<tr>
+						<th>
+							<div class="swsales_reports-sale-value">
+								<a href="<?php echo esc_url( add_query_arg( array( 'post' => $sitewide_sale->get_id(), 'action' => 'edit' ), admin_url( 'post.php' ) ) ); ?>"><?php echo esc_html( $sitewide_sale->get_name() ); ?></a>
+							</div>
+							<div class="swsales_reports-sale-value-description">
+								<?php
+									printf(
+										wp_kses_post( '%s to %s', 'sitewide-sales' ),
```

Comment:
Is this wp_kses_post lazy? I think we had one or more before around this code. Are we okay with ANY HTML in these values?

Reply:
I don't think we want any HTML actually. I can switch this to just esc_html?